feat: add iconset generator (#1612)
vaadin/docs: articles/components/icons/index.asciidoc

The icons are rendered as an inline `<svg>` element inside the shadow root of a `<vaadin-icon>` element, allowing you to style them using CSS.

> == Creating a Custom Icon Set

Reviewer: A nitpick, but is it "Creating" or something like "Using" or "Importing" a custom icon set?

Author: Right, “creating” might indeed give the wrong impression. How about:

```suggestion
== Using a Third-Party Icon Set
```

A collection of SVG files can be converted to the `<vaadin-iconset>` format, which is “consumed” by the `<vaadin-icon>`/`Icon` component.
You can take a look at the https://github.com/vaadin/web-components/blob/master/packages/icons/vaadin-iconset.js[implementation of the Vaadin icon set] for reference.

Optionally, you can create a Java class/enum as a type-safe API for your Flow applications.

=== Icon Set Generator

This tool converts a folder of SVG files into `<vaadin-iconset>` and corresponding Java enum files.

> You can then download them and place them in your project.

Reviewer: This sounds a bit abstract. I personally have some experience creating addons for another icon set. Yet, reading this doesn't make it clear to me where to place those files, nor how to use them in my project. Could we clarify this with some specific instructions?

Author: I try to explain it here: https://github.com/vaadin/docs/pull/1612/files#diff-6173ef28a9a9f1df3042809fc52b9384cd19405135f48a6300b035974795e448R290-R292

Author: For using the icons, perhaps there could be more instructions, but I thought the earlier content on the page would suffice.

Author: I tried to improve this a little bit in https://github.com/vaadin/docs/pull/1612/commits/2054809864f518358b2d99ace1863dbbb6a2a09a

[source,typescript,role="render-only embedded"]
----
include::{root}/frontend/demo/component/icons/iconset-generator.ts[render, persistent]
----

=== Other Icon Formats

> Third-party icon sets come in many different formats, and there is not a single integration method that works for every format.

Reviewer 2: ```suggestion
Third-party icon sets come in many different formats, and there is no single integration method that works for every format.
```
